op-guide/docker.md

```diff
@@ -1,9 +1,196 @@
+  -v /usr/share/ca­certificates/:/etc/ssl/certs \
+  -v /etc/localtime:/etc/localtime:ro \
+  --volumes-from ti-storage \
+  pingcap/pd \
+  --cluster-id=1 \
+  --name="pd3" \
+  --data-dir="/tidata/pd3" \
+  --client-urls="http://0.0.0.0:2379" \
+  --advertise-client-urls="http://pd3:2379" \
+  --peer-urls="http://0.0.0.0:2380" \
+  --advertise-peer-urls="http://pd3:2380" \
+  --initial-cluster="pd1=http://pd1:2380,pd2=http://pd2:2380,pd3=http://pd3:2380" \
+  --addr="0.0.0.0:1234" \
+  --advertise-addr="pd3:1234"
+```
+
+After that, if you need to add **pd4** into the existing cluster, just use `--join` flag, and specify any one of the available **advertise-client-urls** above.
+Notice that the **advertise-client-urls** is needed here, not **advertise-peer-urls**.
+
+**pd4:**
+
+```bash
+docker run --net ${net} -d --name pd4 \
+  -v /usr/share/ca­certificates/:/etc/ssl/certs \
+  -v /etc/localtime:/etc/localtime:ro \
+  --volumes-from ti-storage \
+  pingcap/pd \
+  --cluster-id=1 \
+  --name="pd4" \
+  --data-dir="/tidata/pd4" \
+  --client-urls="http://0.0.0.0:2379" \
+  --advertise-client-urls="http://pd4:2379" \
+  --peer-urls="http://0.0.0.0:2380" \
+  --advertise-peer-urls="http://pd4:2380" \
+  --join="http://pd1:2379" \
+  --addr="0.0.0.0:1234" \
+  --advertise-addr="pd4:1234"
+```
+
+## Start TiKV service
+
+Next you can run any number of TiKV instances, which is the underlying distributed storage.
+
+**tikv1:**
+
+```bash
+docker run --net ${net} -d --name tikv1 \
+  -v /etc/localtime:/etc/localtime:ro \
+  --volumes-from ti-storage \
+  pingcap/tikv \
+  --addr="0.0.0.0:20160" \
+  --advertise-addr="tikv1:20160" \
+  --store="/tidata/tikv1" \
+  --dsn=raftkv \
+  --pd="pd1:2379,pd2:2379,pd3:2379" \
+  --cluster-id=1
+```
+
+**tikv2:**
+
+```bash
+docker run --net ${net} -d --name tikv2 \
+  -v /etc/localtime:/etc/localtime:ro \
+  --volumes-from ti-storage \
+  pingcap/tikv \
+  --addr="0.0.0.0:20160" \
+  --advertise-addr="tikv2:20160" \
+  --store="/tidata/tikv2" \
+  --dsn=raftkv \
+  --pd="pd1:2379,pd2:2379,pd3:2379" \
+  --cluster-id=1
+```
+
+**tikv3:**
+
+```bash
+docker run --net ${net} -d --name tikv3 \
+  -v /etc/localtime:/etc/localtime:ro \
+  --volumes-from ti-storage \
+  pingcap/tikv \
+  --addr="0.0.0.0:20160" \
+  --advertise-addr="tikv3:20160" \
+  --store="/tidata/tikv3" \
+  --dsn=raftkv \
+  --pd="pd1:2379,pd2:2379,pd3:2379" \
+  --cluster-id=1
+```
+
+## Start TiDB service
+
+The **tidb-server** as the SQL Layer is stateless, and accept client connections from users.
+Using `-p 4000:4000` to expose port of 4000 to the host server.
+
+```bash
+docker run --net ${net} -d --name tidb \
+  -p 4000:4000 \
+  -v /etc/localtime:/etc/localtime:ro \
+  pingcap/tidb \
+  --store=tikv \
+  --path="pd1:2379,pd2:2379,pd3:2379?cluster=1" \
+  -L warn
+```
 
+## Use SQL client
+After you started a TiDB cluster, you can use official mysql client connecting to TiDB for a test immediately.
```

Comment:
+After you start a TiDB cluster, you can use the official MySQL client to connect to TiDB for a test immediately.